common/common.cpp

```diff
@@ -156,13 +291,449 @@ static int count_math_cpus(int cpu_count) {
+    if (hSnapshot != INVALID_HANDLE_VALUE) {
+        te.dwSize = sizeof(THREADENTRY32);
+        Thread32First(hSnapshot, &te);
+        if (Thread32Next(hSnapshot, &te)) {
+            do {
+                // Check if the thread is part of this process
+                if (te.th32OwnerProcessID == GetProcessId(hProcess)) {
+                    // Set thread affinity
+                    if (!SetThreadAffinityMask(hThread, cpuMask.to_ullong() & systemAffinityMask)) {
+                        CPUSET_PRINT_DEBUG("Could not set affinity for Main Process Thread\n");
+                    }
+                }
+            } while( Thread32Next(hSnapshot, &te ) );
+        } 
+        CloseHandle(hSnapshot);
+    }
+
+	if (hProcess)
+		::CloseHandle(hProcess);
+	if (hThread)
+		::CloseHandle(hThread);
+
+    HANDLE hProcess2 = ::OpenProcess(PROCESS_ALL_ACCESS, FALSE, GetCurrentProcessId());
+    
+    if (hProcess2) {
+        PROCESS_POWER_THROTTLING_STATE PowerThrottling;
+        RtlZeroMemory(&PowerThrottling, sizeof(PowerThrottling));
+        PowerThrottling.Version = PROCESS_POWER_THROTTLING_CURRENT_VERSION;
+
+        PowerThrottling.ControlMask = PROCESS_POWER_THROTTLING_IGNORE_TIMER_RESOLUTION;
+        PowerThrottling.StateMask = 0;
+        PowerThrottling.StateMask = PROCESS_POWER_THROTTLING_IGNORE_TIMER_RESOLUTION;
+
+        SetProcessInformation(hProcess2,
+            ProcessPowerThrottling,
+            &PowerThrottling,
+            sizeof(PowerThrottling));
+    
+        RtlZeroMemory(&PowerThrottling, sizeof(PowerThrottling));
+        PowerThrottling.ControlMask = PROCESS_POWER_THROTTLING_EXECUTION_SPEED;
+        PowerThrottling.StateMask = 0;
+        PowerThrottling.StateMask = PROCESS_POWER_THROTTLING_EXECUTION_SPEED;
+        SetProcessInformation(hProcess2,
+            ProcessPowerThrottling,
+            &PowerThrottling,
+            sizeof(PowerThrottling));
+
+        MEMORY_PRIORITY_INFORMATION MemPrio;
+        ZeroMemory(&MemPrio, sizeof(MemPrio));
+        MemPrio.MemoryPriority = MEMORY_PRIORITY_NORMAL;
+
+        SetProcessInformation(hProcess2,
+            ProcessMemoryPriority,
+            &MemPrio,
+            sizeof(MemPrio));
+
+		::CloseHandle(hProcess2);
+    }
+    
+    return coreSelected;
+}
+
+ULONG set_procMask(int32_t direction = 0, int32_t req_threads = 0, int32_t lltraversal = 0, int32_t allowtc = 0, int32_t allowcz = 0, int64_t cpuMask = 0) {
+    std::bitset<64> bMask;
+
+    bMask = generate_Mask(direction, req_threads, lltraversal, allowtc, allowcz, cpuMask);
+
+    numPhysicalCores = get_count_procMask(bMask.to_ullong());
+
+    CPUSET_PRINT_DEBUG("Generated Mask: %s\n", bMask.to_string().c_str());
+    return bMask.to_ullong();
+}
+
+#endif // _WIN32
 
 /**
  * Returns number of CPUs on system that are useful for math.
  */
 int get_math_cpu_count() {
-#if defined(__x86_64__) && defined(__linux__)
+#if defined(__x86_164__) && defined(__linux__)
```

Comment:
Is this a typo?

Reply:
It's definitely a typo

Reply:
Correcting it will skip the `#elif` below that `#if` for x86-64 Linux (may or may not be intended).